odl/tomo/data/uncompr_bin.py

```diff
@@ -0,0 +1,716 @@
+    written directly using `write_data`, which allows to write arbitrary
+    portions.
+    """
+
+    def __init__(self, file, header=OrderedDict()):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        file : file-like or str
+            Stream or filename to which to write the data. The stream
+            is allowed to be already opened in a writable mode.
+        header : `OrderedDict`, optional
+            Header in form of an ordered dictionary, where each entry has
+            the following form::
+
+                'name': {'value': value_as_array,
+                         'offset': offset_in_bytes
+                         'description': description_string}
+
+            All ``'value'``'s must be `numpy.ndarray`'s. Their size and the
+            ``'offset'`` entry determine the space that the value occupies
+            in `file`'s header.
+
+            If an empty dictionary is given, no header is written.
+
+        Notes
+        -----
+        **Important:** There is no check if the writes from the provided
+        header are consistent, i.e., that they don't overwrite each other.
+        Ensuring this is the user's responsibility.
+
+        See Also
+        --------
+        header_fields_from_table :
+            Function to parse a specification table, returning a field
+            sequence usable as ``header_fields`` parameter.
+        FileReaderRawBinaryWithHeader.read_header
+        """
+        # Need those attrs in subsequent code
+        file_attrs = ('mode', 'seek', 'write', 'close')
+        is_file = all(hasattr(file, attr) for attr in file_attrs)
+        if is_file:
+            self.__file = file
+            self.__file_shared = True
+        else:
+            self.__file = open(file, 'wb')
+            self.__file_shared = False
+
+        if 'b' not in self.file.mode:
+            raise ValueError("`file` must be opened in binary mode, "
+                             "but mode 'is {}'".format(self.file.mode))
+
+        if not isinstance(header, dict):
+            raise TypeError('`header` must be a dictionary, got {!r}'
+                            ''.format(header))
+        self.__header = header
+
+    @property
+    def file(self):
+        """File object from which ``self`` reads."""
+        return self.__file
+
+    def __enter__(self):
+        """Initializer for the context manager."""
+        return self
+
+    def __exit__(self, *exc):
+        """Cleanup before on exiting the context manager."""
+        if not self.__file_shared:
+            self.file.close()
+
+    @property
+    def header_size(self):
+        """Size of `header` in bytes."""
+        if not self.header:
+            return 0
+
+        # Determine header size by finding the largest offset and the
+        # value of the corresponding entry. The header size is the
+        # offset plus the size of the entry.
+        max_offset = 0
```

Comment:
Here you basically re-implement max. Why not something like

```python
max_key = max(self.header, key=lambda key: self.header[key]['offset'])
max_offset = self.header[max_key]['offset']
max_value = self.header[max_key]['value']
```

also makes it more obvious what you're aiming at.